mne/decoding/receptive_field.py

```diff
@@ -121,17 +121,13 @@ def _delay_and_reshape(self, X, y=None):
         """Delay and reshape the variables."""
         if not isinstance(self.estimator_, TimeDelayingRidge):
             # X is now shape (n_times, n_epochs, n_feats, n_delays)
-            X_del = _delay_time_series(X, self.tmin, self.tmax, self.sfreq,
-                                       newaxis=X.ndim,
-                                       fill_mean=self.fit_intercept)
-        else:
-            X_del = X[..., np.newaxis]
-
-        X_del = _reshape_for_est(X_del)
-        # Concat times + epochs
-        if y is not None:
-            y = y.reshape(-1, y.shape[-1], order='F')
-        return X_del, y
+            X = _delay_time_series(X, self.tmin, self.tmax, self.sfreq,
+                                   fill_mean=self.fit_intercept)
+            X = _reshape_for_est(X)
+            # Concat times + epochs
+            if y is not None:
```

Comment:
just confirming that you want `y is not None` to be inside the `if` statement, since before it was outside the `if` statement

Reply:
Yep, `TimeDelayingRidge` is now epochs-aware